src/headers/switcher-data-structs.hpp

```diff
@@ -201,18 +200,34 @@ struct IdleData
 	bool alreadySwitched = false;
 };
 
-typedef enum {
-	NO_SWITCH = 0,
-	SWITCH = 1,
-	RANDOM_SWITCH = 2
-} NoMatch;
+struct MediaSwitch {
+	OBSWeakSource scene;
+	OBSWeakSource source;
+	OBSWeakSource transition;
+	obs_media_state state;
+	uint64_t time;
+	time_restriction restriction;
+	bool matched;
+
+	inline MediaSwitch(OBSWeakSource scene_, OBSWeakSource source_,
+			   OBSWeakSource transition_, obs_media_state state_,
+			   enum time_restriction restriction_, uint64_t time_)
```

Comment:
First of all, thanks for this PR! Looks good!

However it only seems to compile for me when removing the "enum" on this line.
```suggestion
			   time_restriction restriction_, uint64_t time_)
```

I assume most of the changes not related to media are formatting changes, right?

Otherwise I had no issues compiling and testing with a small video file.